app/models/test_track/session.rb

```diff
@@ -31,23 +31,41 @@ def state_hash
     }
   end
 
-  def log_in!(identifier_type, identifier_value, opts = {})
+  def log_in!(identity, opts = {})
```

Comment:
This is the breaking change, but I think its a much nicer contract:

`session.log_in!('user_id', user.id)` vs `session.log_in!(user)`

Of course, this requires that User mixes in `TT::Identity`.

Reply:
and we're relying on conformance to the duck type of responding to `test_track_identifier_type` and `test_track_identifier_value` as our verification of mixing in `TT::Identity`? That would mean that a mistake would result in a method missing error rather, right? Should we have it check for inheritance from `TT::Identity` and throw a more actionable error message?

Reply:
I was debating adding a check. I left it out because I thought relying on duck typing was more ruby-esque. I can add it tho. For this new feature to work, you do need to pass in something that inherits from `TT::Identity`, so we might as well require it.

Reply:
I think it would still be sufficiently rubyish to check if it `responds_to?` the required methods or raise.